dtfs-central-api/src/mapping/mapUtilisationReport.ts

```diff
@@ -20,12 +10,48 @@ const mapAzureFileInfoEntityToAzureFileInfo = (fileInfoEntity: AzureFileInfoEnti
   mimetype: fileInfoEntity.mimetype,
 });
 
-export const mapUtilisationReportEntityToGetUtilisationReportResponse = (reportEntity: UtilisationReportEntity): GetUtilisationReportResponse => ({
-  bankId: reportEntity.bankId,
-  id: reportEntity.id,
-  status: reportEntity.status,
-  uploadedByUserId: reportEntity.uploadedByUserId,
-  reportPeriod: mapReportPeriodEntityToReportPeriod(reportEntity.reportPeriod),
-  azureFileInfo: reportEntity.azureFileInfo ? mapAzureFileInfoEntityToAzureFileInfo(reportEntity.azureFileInfo) : null,
-  dateUploaded: reportEntity.dateUploaded,
-});
+function assertAllUploadFieldsArePopulated(uploadedReportEntity: UtilisationReportEntity): asserts uploadedReportEntity is UtilisationReportEntity & {
+  dateUploaded: Date;
+  azureFileInfo: AzureFileInfoEntity;
+  uploadedByUserId: string;
+} {
+  const allUploadFieldsArePopulated: boolean =
+    uploadedReportEntity.dateUploaded !== null && uploadedReportEntity.azureFileInfo !== undefined && uploadedReportEntity.uploadedByUserId !== null;
+  if (!allUploadFieldsArePopulated) {
+    throw new Error('Failed to map data - report seems to have been uploaded but is missing some required fields');
+  }
+}
+
+export const mapUtilisationReportEntityToGetUtilisationReportResponse = async (
```

Comment:
I was perhaps being a bit cheeky not adding unit tests for this specifically the first time around, but now that this contains a fair amount more logic I think we should introduce unit tests